### DISCUSSION TITLE
Sending of '2' as ping to the server

### DISCUSSION THREAD

**Discussion by Notadumdum** [socketio/socket.io#4650] [Q&A]
Does client version 2.5.0 automatically send '2' as ping to the server or it has to be stated on the code written in the client side

**Comment by darrachequesne** [comment]
Hi! Yes, the client will automatically send a PING packet to the server, based on the `pingInterval` value.

Reference: https://socket.io/docs/v4/how-it-works/#disconnection-detection

**Reply by Notadumdum** [reply]
Version 2.5.0 not version 4 please?

**Reply by darrachequesne** [reply]
Wrong link, sorry. The "How it works" page does not exist for version 2.x, so yes, in version 2.5.0 the client will automatically send a PING packet to the server.

In v4, it's the server that sends the PING packets.
